proc-macros/src/helpers.rs

```diff
@@ -165,6 +165,20 @@ pub(crate) fn is_option(ty: &syn::Type) -> bool {
 	false
 }
 
+/// Iterates over all Attribute's and parses only the attributes that are doc comments.
+///
+/// Note that `doc comments` are expanded into `#[doc = "some comment"]`
+/// Thus, if the attribute starts with `doc` => it's regarded as a doc comment.
+pub(crate) fn extract_doc_comments(attrs: &[syn::Attribute]) -> Vec<syn::Attribute> {
+	attrs
+		.iter()
+		.filter_map(|attr| match attr.path.segments.first() {
```

Comment:
In `scale-info` this is done in a slightly different way: https://github.com/paritytech/scale-info/blob/45bee5967b0fb1b282f326ffb1aa02ceddac7834/derive/src/lib.rs#L264 – might be a little more robust? Not sure if we need to strip leading whitespace here though.

Reply:
replaced it with the scale-info code without stripping